### PR TITLE
Feat/diff palette

### PR DESCRIPTION
**Pull request by eh-am**
https://user-images.githubusercontent.com/6951209/150017644-0187e2de-aa91-42f9-ad0c-18f0cca49d3a.mp4

Related to #657 

Comments:
* Ended up doing for diff only, altough created a FlamegraphPalettes abstraction that apply to ALL flamegraphs. 
* went with the naive approach and lifted the state up to `FlamegraphRenderer`, so you gonna see a lot of `palette` being passed around. In a future refactor we can use [Context](https://reactjs.org/docs/context.html)
* not persisting the preference to local storage

### CODE REVIEW

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/Flamegraph_render.ts [hunk 1]
```diff
@@ -79,6 +82,7 @@ export default function RenderCanvas(props: CanvasRendererConfig) {
   const { units } = props;
   const { rangeMin, rangeMax } = props;
   const { tickToX } = props;
+  const { palette } = props;
```
Comment (Rperry2174) [inline webapp/javascript/components/FlameGraph/FlameGraphComponent/Flamegraph_render.ts:85]: just curious, could all these be combined into one? Any particular reason they are not? 

i.e. 
```
const {
    canvas,
    numTicks,
    sampleRate ,
    pxPerTick,
    fitMode,
    units,
    rangeMin ,
    rangeMax,
    tickToX ,
    palette
} = props;
```

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/colorPalette.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Color.rgb(128, 110, 183),
+  ],
+};
+
+export const ColorBlindPalette: FlamegraphPalette = {
+  ...DefaultPalette,
+
+  name: 'Color Blind',
+  goodColor: Color('rgb(26,133,255)'),
+  neutralColor: Color.rgb(148, 142, 142),
```
Comment (Rperry2174) [inline webapp/javascript/components/FlameGraph/FlameGraphComponent/colorPalette.ts:56]:
```
  goodColor: Color('rgb(26,133,255)'),
  neutralColor: Color.rgb(148, 142, 142),
```
why is one in quotes and the other is not?

**Comment by Rperry2174** [comment]
Hey @eh-am this looks great so far... can you elaborate on this:

> went with the naive approach and lifted the state up to FlamegraphRenderer, so you gonna see a lot of palette being passed around. In a future refactor we can use Context


1. Why did you choose this approach?
2. As a followup to this PR could you write more details on what that would look like to switch to using Context?

**Comment by eh-am** [comment]
image
The tooltip should use the same palette

File: webapp/javascript/components/FlameGraph/FlameGraphComponent/color.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 /* eslint-disable camelcase */
 import Color from 'color';
+import { scaleThreshold, scaleLinear } from 'd3-scale';
 import murmurhash3_32_gc from './murmur3';
+import type { FlamegraphPalette } from './colorPalette';
 
 const colors = [
```
Comment (Rperry2174) [inline webapp/javascript/components/FlameGraph/FlameGraphComponent/color.ts:7]: do we still need this array here now?

Comment (Rperry2174) [reply webapp/javascript/components/FlameGraph/FlameGraphComponent/color.ts:7]: nvm will address in future PR

**Pull request merged**